sites/shared/components/workbench/edit.js

```diff
@@ -0,0 +1,143 @@
+import yaml from 'js-yaml'
+import defaultSettings from './default-settings'
+import validateGist from './gist-validator'
+import { useEffect, useState, useRef } from 'react'
+import Popout from 'shared/components/popout.js'
+import { useTranslation } from 'next-i18next'
+import { capitalize } from '@freesewing/core'
+
+/** count the number of lines in some text */
+const countLines = (txt) => txt.split('\n').length
+
+/** a view for editing the gist as yaml */
+const Edit = (props) => {
+  let { gist, setGist, gistReady } = props
+
+  const inputRef = useRef(null)
+  // the gist parsed to yaml
+  const [gistAsYaml, setGistAsYaml] = useState(null)
+  // the number of lines in the yaml
+  const [numLines, setNumLines] = useState(0)
```

Comment:
Do we care about the line numbers?
By which I mean, are we going to say things like: One line X you are missing blabla?

Because if we just want to show them, I'd prefer to handle that in CSS.

Reply:
I can add that. It's not worth spending much time on :)